zone/zonedb.cpp

```diff
@@ -994,6 +1040,55 @@ bool ZoneDatabase::SaveCharacterLootLockout(uint32 character_id, uint32 expiry,
 	return true;
 }
 
+bool ZoneDatabase::SaveCharacterInstanceLockout(uint32 character_id, uint32 expiry, uint32 zone_id, uint32 zone_instance_id)
+{
+	std::string query = StringFormat("REPLACE INTO `character_instance_lockouts` (character_id, expiry, zone_id, zone_instance_id) VALUES (%u, %u, %u, %u)", character_id, expiry, zone_id, zone_instance_id);
+	auto results = QueryDatabase(query);
+	return true;
+}
+
+int32 ZoneDatabase::GetHighestZoneInstanceID(uint32 zone_id)
+{
+	std::string query = StringFormat("SELECT MAX(zone_instance_id) FROM `character_instance_lockouts` WHERE (zone_id = %u)", zone_id);
```

Comment:
You might consider adding an index for `zone_id` for this query? Not sure how big this table is expected to get, but the index on the two columns currently won't apply. it'll only apply to the first one in the double key if used alone

Reply:
Done - on the server side. Will be in next dump